### PR TITLE
Roland DJ-505: Fix Pad Mode Switch Shift state

### PR DESCRIPTION
**Pull request by Holzhaus**
This expects the `isShifted` attribute to be set correctly on the pad section `ComponentContainer` and therefore depends on PR #2266.

### CODE REVIEW

**Comment by Holzhaus** [comment]
This can lead to deleting hotcues by accident, so I'd like to get to get this merged ASAP.

**Comment by Holzhaus** [comment]
@Be-ing ping

File: res/controllers/Roland_DJ-505-scripts.js [hunk 1]
```diff
@@ -1094,59 +1094,63 @@ DJ505.PadSection.prototype.padModeButtonPressed = function (channel, control, va
...
 };
 
 DJ505.PadSection.prototype.setPadMode = function (control) {
-    if (this.padMode === control) {
-        return;
-    }
-
     var mode = this.controlToPadMode(control);
-    if (mode === undefined) {
+    if (this.currentMode === mode || mode === undefined) {
```
Comment (Be-ing) [inline res/controllers/Roland_DJ-505-scripts.js]: Should this be
```suggestion
    if (this.currentMode === mode || this.currentMode === undefined) {
```
?

File: res/controllers/Roland_DJ-505-scripts.js [hunk 2]
```diff
@@ -1094,59 +1094,63 @@ DJ505.PadSection.prototype.padModeButtonPressed = function (channel, control, va
...
+        // CUE mode pads become shifted.
+        // When you're in CUE LOOP mode and want to switch back to
+        // HOT CUE mode, the user need to press HOT CUE (without holding
+        // SHIFT). However, the HOT CUE mode pads are still shifted even though
+        // the user is not pressing [SHIFT] because they never got the unshift
+        // event (the [SHIFT] button was released in CUE LOOP mode, not in HOT
+        // CUE mode).
+        // Hence, it's necessary to set the correct shift state when switching
+        // modes.
+        if(this.isShifted) {
```
Comment (Be-ing) [inline res/controllers/Roland_DJ-505-scripts.js]:
```suggestion
        if (this.isShifted) {
```

File: res/controllers/Roland_DJ-505-scripts.js (hunk 1 above)
Comment (Holzhaus) [reply res/controllers/Roland_DJ-505-scripts.js]: No. `mode` is the *new* mode. If it's `undefined`, then the user pressed a mode button that is currently not mapped (e.g. `FLIP` mode, `SLICER` mode, etc.). In that case, nothing happens and the current mode stays active. I'll add a commit that clears this up by using better variable names.

File: res/controllers/Roland_DJ-505-scripts.js (hunk 2 above)
Comment (Holzhaus) [reply res/controllers/Roland_DJ-505-scripts.js]: ok

**Comment by Holzhaus** [comment]
I added some minor improvements and comments to the code. Feel free to merge.

**Comment by Be-ing** [comment]
Thank you for taking care to fix the library in #2266 first instead of hacking around it in your controller script.

**Pull request merged**